File: WorkScheduler.API/Models/Job.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WorkScheduler.API.Models
{
    public class Job
    {
        public int Id { get; set; }
        public string JobNumber { get; set; }
        public int Visit { get; set; }
        public string PayerType { get; set; }
        public string ProblemGiven { get; set; }
        public DateTime DateAssigned { get; set; }
        public DateTime TimeFrom { get; set; }
        public DateTime TimeTo { get; set; }
        public string Address { get; set; }
        public string PostCode { get; set; }
        public string Report { get; set; }
        public bool slotReplaced { get; set; }
        public int? slotIndex { get; set; }
        public bool Key { get; set; }
        public string KeyAddress { get; set; }
        public string AgencyReference { get; set; }
        public string LandlordName { get; set; }
        public string LandlordPhone { get; set; }
        public string TenantName { get; set; }
        public string TenantPhone { get; set; }
        public string PrivateName { get; set; }
        public string PrivatePhone { get; set; }
        public string AgencyContactName { get; set; }
        public string AgencyPhone { get; set; }
        public int? AgencyId { get; set; }
        public Agency Agency { get; set; }
        public ApplianceType ApplianceType { get; set; }
        public virtual ICollection<Photo> Photos { get; set; }
        public virtual ICollection<JobTag> JobTags { get; set; }
    }
}

File: WorkScheduler-SPA/src/app/_models/Job.ts
import { Agency } from './Agency';
import { Tag } from './Tag';
import { Photo } from './Photo';
import { ApplianceType } from './ApplianceType';

export interface Job {
    id: number;
    jobNumber: string;
    visit: number;
    report?: string;
    visitToDisplay?: string;
    payerType: string;
    problemGiven?: string;
    dateAssigned: Date;
    timeFrom: Date;
    timeTo: Date;
    slotReplaced: boolean;
    slotIndex?: number;
    key?: boolean;
    keyAddress?: string;
    address?: string;
    postCode?: string;
    agencyReference?: string;
    landlordName?: string;
    landlordPhone?: string;
    privateName?: string;
    privatePhone?: string;
    tenantName?: string;
    tenantPhone?: string;
    agency?: Agency;
    applianceType?: ApplianceType;
    agencyContactName: string;
    agencyPhone: string;
    tags: Tag[];
    photos: Photo[];
}
